torch/csrc/jit/passes/dead_code_elimination.cpp

```diff
@@ -259,18 +259,49 @@ class DeadCodeEliminator {
     return has_side_effects;
   }
 
+  void removeIfNodeOutput(Node * node, size_t i) {
+    node->eraseOutput(i);
+    for (Block* b : node->blocks()) {
+      b->eraseOutput(i);
+    }
+  }
+
   void removeDeadBlockOutputs(Node* node) {
     if (node->kind() != prim::If && node->kind() != prim::GradOf) {
       return;
     }
 
     for (size_t i_1 = node->outputs().size(); i_1 > 0; --i_1) {
       size_t i = i_1 - 1;
-      if (!node->outputs().at(i)->hasUses()) {
-        node->eraseOutput(i);
-        for (Block* b : node->blocks()) {
-          b->eraseOutput(i);
+      if (node->kind() == prim::If) {
```

Comment:
I'm wary of special-casing this code too much. It seems like the case this is trying to handle (assignment in the else-block of an `if not None` check) is unlikely to appear in user code. I think we should just leave the unwrap_optional's in for those cases—if it turns out to be a common pattern somehow, we can revisit these additions.

Reply:
There were jitter issues at one point but unrelated to this. This is just trying to remove unneeded if outputs.